tests/integration/models/api-key.spec.coffee

```diff
@@ -0,0 +1,136 @@
+_ = require('lodash')
+m = require('mochainon')
+Promise = require('bluebird')
+
+{ resin, givenLoggedInUser } = require('../setup')
+
+eventuallyExpectProperty = (promise, prop) ->
+	m.chai.expect(promise).to.eventually.have.property(prop)
+
+describe 'API Key model', ->
+
+	givenLoggedInUser()
+
+	describe 'resin.models.apiKey.create()', ->
+
+		it 'should be able to create a new api key', ->
+			resin.models.apiKey.create('apiKey')
+			.then (key) ->
+				m.chai.expect(key).to.be.a('string')
+
+		it 'should be able to create a new api key with description', ->
+			resin.models.apiKey.create('apiKey', 'apiKeyDescription')
+			.then (key) ->
+				m.chai.expect(key).to.be.a('string')
+				resin.models.apiKey.getAll()
+			.then (apiKeys) ->
+				m.chai.expect(apiKeys).to.be.an('array')
+				m.chai.expect(apiKeys).to.have.lengthOf(1)
+				m.chai.expect(apiKeys[0]).to.have.property('name', 'apiKey')
+				m.chai.expect(apiKeys[0]).to.have.property('description', 'apiKeyDescription')
```

Comment:
You could simplify this (and most other assertions here) to just:

```coffee
m.chai.expect(apiKeys).to.deep.equal([
  { name: 'apiKey', description: 'apiKeyDescription' }
])
```

I think that's equivalent to what you've got here, but shorter & clearer.

Reply:
Ah, except presumably you get an id property, and that gets tricky.

There's probably still a nice similar solution with some kind of matching plugin (e.g. https://www.npmjs.com/package/chai-deep-match).

Alternatively, it's a bit of a hack, but I think this should work:

```coffee
m.chai.expect(apiKeys).to.deep.equal([
  { id: apiKeys[0].id, name: 'apiKey', description: 'apiKeyDescription' }
]);
```

Or maybe even

```coffee
m.chai.expect(
  _.map(apiKeys, (k) => _.omit(k, 'id')
).to.deep.equal([
  { name: 'apiKey', description: 'apiKeyDescription' }
])
```

Reply:
I was thinking about using that package for a long time.
I guess that the time to add it to our dependencies has come.
We definitely need this in the billing tests as well.